api/src/main/resources/moduleApplicationContext.xml

```diff
@@ -136,4 +136,13 @@
 
     <bean id="registrationcore.openEmpiPatientFilter"
           class="org.openmrs.module.registrationcore.api.mpi.openempi.OpenEmpiPatientFilter"/>
+          
+    <!-- Add global property listeners -->
+	<bean id="registrationcore.globalPropertyListeners" parent="openmrsEventListeners">
+	    <property name="globalPropertyListeners">
+	        <list value-type="org.openmrs.api.GlobalPropertyListener" merge="true">
+	            <bean class="${project.parent.groupId}.${project.parent.artifactId}.api.impl.RegistrationCoreServiceImpl" />
+	        </list>
+	    </property>
+	</bean>
```

Comment:
Ok this is coupled with implementing [`RegistrationCoreServiceImpl.globalPropertyChanged(GlobalProperty)`](https://github.com/openmrs/openmrs-module-registrationcore/blob/d9596396ae7d822584dc56c24ea739a35c9a1315/api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java#L249-L252) correct?

If yes, this should be part of another PR I think?

Reply:
We need this even before https://issues.openmrs.org/browse/RC-32 . I trust initial authors just forgot to add it, does it really need creation of a new ticket?

Reply:
Well if you hook the GP listener then you will trigger the below behaviour each time a GP is updated no?
```java
@Override
public void globalPropertyChanged(GlobalProperty gp) {
  idSource = null;
}
```
... and that's unwanted to say the least.

I think a new ticket would be best to handle an appropriate implementation of what should happen when a specific GP's value is changed.

Reply:
As I think and know, that implementation is appropriate, Even at runtime. I'm sure that implementation is perfect. The only missing thing was adding this config. 

I added this here cause of my test unit. Yes we can leave it out but it's so so crucial. We either create a new ticket and add prior to merging this or let's leave it here @mks-d

Reply:
@samuelmale my understanding is this that, by enabling the GP listener on this class, the above code will be run each time a GP is edited. In other words `idSource` will be set to null as soon as _any_ GP is edited. Are we sure that this is what we want?

I would assume that if you hook the GP listener to that class, then you ought to provide an appropriate implementation for `RegistrationCoreServiceImpl.globalPropertyChanged(GlobalProperty)`. It should check whether the GP that is ediited is `"registrationcore.identifierSourceId"`, and if it is, it should update `idSource` accordingly.

Reply:
Makes sense. I think that could come after this. Maybe as another ticket! It's actually the right thing. We want to set the `idSource = null` when and only the `registrationcore.identifierSourceId` is edited. This is because the `idSource` is a static member so if the `identifierSource` is changed using the GP the current one will need to be set to null so that the new one is picked up [here](https://github.com/openmrs/openmrs-module-registrationcore/blob/d9596396ae7d822584dc56c24ea739a35c9a1315/api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java#L490-L497). Unless we have the bean above, this won't be possible at runtime once the identifier source was already initialized in the `RegistrationCoreService`.

As for now, I believe the current implementation still works anyway! But anyway, we could create a different ticket for a better implementation of `RegistrationCoreServiceImpl.globalPropertyChanged(GlobalProperty)`  @mks-d

Reply:
@samuelmale ok that's a bit weird, I would rather have a code that ensures that `idSource` is set at the value that it should, rather than having another method safe guarding against the null value.

@mseaton did you get a chance to look at @samuelmale's PR here?